master/buildbot/schedulers/base.py

```diff
@@ -412,8 +412,8 @@ def addBuildsetForSourceStampSetDetails(self, reason, sourcestamps,
                                                           triggeredbybrid)
 
                 if brids and brids.keys() == builderNames:
+                    log.msg("Resuing existing buildrequest %s for trigger by %s" % (brids, triggeredbybrid))
                     defer.returnValue((bsid, brids))
-                    return
```

Comment:
@tdctaz this changed seems unintentional I think we should undo it

Reply:
@tdctaz as we talked we are going to remove this logic, which can cause issues from line 398-415 and also remove getBuildRequestsTriggeredByScheduler which incorrectly filters by property